src/cis0.rs

```diff
@@ -0,0 +1,251 @@
+//! This module contains types and functions for interacting with smart
+//! contracts following the [CIS-0](https://proposals.concordium.software/CIS/cis-0.html) specification.
+
+use crate::{
+    types::{self as sdk_types, smart_contracts::ContractContext},
+    v2::{BlockIdentifier, QueryResponse},
+};
+use concordium_base::{
+    base::Energy,
+    contracts_common::{Amount, ContractName, EntrypointName, OwnedReceiveName, ParseError},
+};
+use sdk_types::{smart_contracts, ContractAddress};
+use smart_contracts::concordium_contracts_common as contracts_common;
+use std::convert::{From, TryFrom};
+use thiserror::*;
+
+/// The query result type for whether a smart contract supports a standard.
+#[derive(Debug)]
+pub enum SupportResult {
+    /// The standard is not supported.
+    NoSupport,
+    /// The standard is supported by the current contract address.
+    Support,
+    /// The standard is supported by using another contract address.
+    SupportBy(Vec<ContractAddress>),
+}
+
+impl SupportResult {
+    /// Return whether the result is [`Support`](Self::Support) or not.
+    pub fn is_support(&self) -> bool { matches!(self, &Self::Support) }
+}
+
+impl contracts_common::Serial for SupportResult {
```

Comment:
You could derive this instead.

Reply:
Well technically yes, but in a hacky way. The contracts-common derive works either in a crate where you have access to concordium_rust_sdk or concordium_std.

I can derive **if** I add the following

```
use crate as concordium_rust_sdk
```
to the module where I derive serialization. But this to me looks very very ugly and I would prefer not to do it.